src/js/bootstrap.js

```diff
@@ -0,0 +1,4153 @@
+        e = t.getBoundingClientRect();
+        var n = q(t, "top"),
+          i = q(t, "left");
+        (e.top += n), (e.left += i), (e.bottom += n), (e.right += i);
+      } else e = t.getBoundingClientRect();
+    } catch (t) {}
+    var o = {
+        left: e.left,
+        top: e.top,
+        width: e.right - e.left,
+        height: e.bottom - e.top,
+      },
+      r = "HTML" === t.nodeName ? U(t.ownerDocument) : {},
+      s = r.width || t.clientWidth || o.width,
+      a = r.height || t.clientHeight || o.height,
+      l = t.offsetWidth - s,
+      c = t.offsetHeight - a;
+    if (l || c) {
+      var u = A(t);
+      (l -= Q(u, "x")), (c -= Q(u, "y")), (o.width -= l), (o.height -= c);
+    }
+    return K(o);
+  }
+  function $(t, e) {
+    var n = arguments.length > 2 && void 0 !== arguments[2] && arguments[2],
+      i = F(10),
+      o = "HTML" === e.nodeName,
+      r = G(t),
+      s = G(e),
+      a = x(t),
+      l = A(e),
+      c = parseFloat(l.borderTopWidth, 10),
+      u = parseFloat(l.borderLeftWidth, 10);
+    n && o && ((s.top = Math.max(s.top, 0)), (s.left = Math.max(s.left, 0)));
+    var h = K({
+      top: r.top - s.top - c,
+      left: r.left - s.left - u,
+      width: r.width,
+      height: r.height,
+    });
+    if (((h.marginTop = 0), (h.marginLeft = 0), !i && o)) {
+      var f = parseFloat(l.marginTop, 10),
+        d = parseFloat(l.marginLeft, 10);
+      (h.top -= c - f),
+        (h.bottom -= c - f),
+        (h.left -= u - d),
+        (h.right -= u - d),
+        (h.marginTop = f),
+        (h.marginLeft = d);
+    }
+    return (
+      (i && !n ? e.contains(a) : e === a && "BODY" !== a.nodeName) &&
+        (h = H(h, e)),
+      h
+    );
+  }
+  function J(t) {
+    var e = arguments.length > 1 && void 0 !== arguments[1] && arguments[1],
+      n = t.ownerDocument.documentElement,
+      i = $(t, n),
+      o = Math.max(n.clientWidth, window.innerWidth || 0),
+      r = Math.max(n.clientHeight, window.innerHeight || 0),
+      s = e ? 0 : q(n),
+      a = e ? 0 : q(n, "left"),
+      l = {
+        top: s - i.top + i.marginTop,
+        left: a - i.left + i.marginLeft,
+        width: o,
+        height: r,
+      };
+    return K(l);
+  }
+  function Z(t) {
+    var e = t.nodeName;
+    if ("BODY" === e || "HTML" === e) return !1;
+    if ("fixed" === A(t, "position")) return !0;
+    var n = I(t);
+    return !!n && Z(n);
+  }
+  function tt(t) {
+    if (!t || !t.parentElement || F()) return document.documentElement;
+    for (var e = t.parentElement; e && "none" === A(e, "transform"); )
+      e = e.parentElement;
+    return e || document.documentElement;
+  }
+  function et(t, e, n, i) {
+    var o = arguments.length > 4 && void 0 !== arguments[4] && arguments[4],
+      r = { top: 0, left: 0 },
+      s = o ? tt(t) : B(t, j(e));
+    if ("viewport" === i) r = J(s, o);
+    else {
+      var a = void 0;
+      "scrollParent" === i
+        ? "BODY" === (a = x(I(e))).nodeName &&
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/hello_100/jca-ui-kit/pullRequest?prid=5731813)